banzai/main.py

```diff
@@ -26,6 +26,49 @@
 
 logger = logs.get_logger(__name__)
 
+ordered_stages = [munge.DataMunger,
+                  qc.ThousandsTest,
+                  qc.SaturationTest,
+                  bias.OverscanSubtractor,
+                  crosstalk.CrosstalkCorrector,
+                  gain.GainNormalizer,
+                  mosaic.MosaicCreator,
+                  bpm.BPMUpdater,
+                  trim.Trimmer,
+                  bias.BiasSubtractor,
+                  dark.DarkSubtractor,
+                  flats.FlatDivider,
+                  photometry.SourceDetector,
+                  astrometry.WCSSolver,
+                  pointing.PointingTest]
+
+
+def get_stages_todo(last_stage, extra_stages=[]):
```

Comment:
This isn't going to do what you want. The [] is evaluated at compile time, not on each function invocation. This is called the "mutable default arguments" problem. Best practice looks clunky, but recommended is

def get_stages_todo(last_stage, extra_stages=None):
    if not extra_stages:
        extra_stages = []